### PR TITLE
Remove questions that do not use StringPools, add StringPools where needed and remove the support of the old version of the Questions

### PR DESCRIPTION
**Pull request by nbaehler**
Use only questions that do not store the titles and texts directly any more. Instead add `StringPools` where needed. Additionally I unified the usage of quiz titles across the app. Every user should now have the same quizzes in the `AuthService` and in the `DBService`.

I had to add a `StringPool` to the `QuizViewModel`.

### CODE REVIEW

File: app/src/main/java/ch/epfl/qedit/view/quiz/QuestionFragment.java [hunk 1]
```diff
@@ -44,12 +44,7 @@ public View onCreateView(
                 .getFocusedQuestion()
                 .observe(
                         getViewLifecycleOwner(),
-                        new Observer<Integer>() {
```
Comment (brunnerant) [inline app/src/main/java/ch/epfl/qedit/view/quiz/QuestionFragment.java:47]: I like when your replace those ugly anonymous classes, thank you !

Comment (nbaehler) [reply app/src/main/java/ch/epfl/qedit/view/quiz/QuestionFragment.java:47]: Oh yes, I missed this one, thx!

**Comment by nbaehler** [comment]
We were a bit too fast on this PR. In the part were we simply answer quizzes we actually do not need to have a `StringPool`. @brunnerant had prepared a function `instantiateLanguage(StringPool pool)` (I had totally forgot it) that replaces the keys by their values in the quiz, questions, ... etc. recursively. Hence I will unwind the changes that I made in this part and make it work with this function. Sorry for the trouble!

**Comment by nbaehler** [comment]
So, I simply took the version on master of the 3 files that handle the answering of quizzes part and added several `instantiateLanguage` in the tests. Works like a charm! Once the CI terminates I will merge this PR.

**Pull request merged**